experimental/native_file_system/virtual_root_provider_win.cc

```diff
@@ -1,12 +1,39 @@
-// Copyright (c) 2014 Intel Corporation. All rights reserved.
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
-
-#include "xwalk/experimental/native_file_system/virtual_root_provider.h"
-
-#include "base/logging.h"
-
-VirtualRootProvider::VirtualRootProvider() {
-  // TODO(darktears): Windows support to be added.
-  NOTIMPLEMENTED();
-}
+// Copyright (c) 2014 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "xwalk/experimental/native_file_system/virtual_root_provider.h"
+
+#include "base/files/file_path.h"
+#include "base/files/file_util.h"
+
+bool VirtualRootProvider::testing_enabled_ = false;
+
+VirtualRootProvider::VirtualRootProvider() {
+  if (!virtual_root_map_.empty())
+    return;
+
+  if (testing_enabled_) {
+    base::GetTempDir(&home_path_);
+  }
+  else {
+    home_path_ = base::GetHomeDir();
+  }
+
+  virtual_root_map_["DESKTOP"] = home_path_.Append(FILE_PATH_LITERAL("Desktop"));
```

Comment:
Yep in the linux provider, however this one should be rewritten to use xdg to get these names. Not sure it's worth it at the moment. I think overall the native file system access needs a lot of love.